src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,641 @@
+        ID3MPlan plan;   // How we calculate target debt
+        uint256  tau;    // Time until you can write off the debt [sec]
+        uint256  culled; // Debt write off triggered
+        uint256  tic;    // Timestamp when the d3m can be culled (tau + timestamp when caged)
+    }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event File(bytes32 indexed ilk, bytes32 indexed what, address data);
+    event File(bytes32 indexed ilk, bytes32 indexed what, uint256 data);
+    event Wind(bytes32 indexed ilk, uint256 amount);
+    event Unwind(bytes32 indexed ilk, uint256 amount, uint256 fees);
+    event Reap(bytes32 indexed ilk, uint256 amt);
+    event Cage(bytes32 indexed ilk);
+    event Cull(bytes32 indexed ilk, uint256 ink, uint256 art);
+    event Uncull(bytes32 indexed ilk, uint256 wad);
+    event Quit(bytes32 indexed ilk, address indexed usr);
+    event Exit(bytes32 indexed ilk, address indexed usr, uint256 amt);
+
+    /**
+        @dev sets msg.sender as authed.
+        @param vat_     address of the DSS vat contract
+        @param daiJoin_ address of the DSS Dai Join contract
+    */
+    constructor(address vat_, address daiJoin_) {
+        vat = VatLike(vat_);
+        daiJoin = DaiJoinLike(daiJoin_);
+        TokenLike(DaiJoinLike(daiJoin_).dai()).approve(daiJoin_, type(uint256).max);
+        VatLike(vat_).hope(daiJoin_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /// @notice Modifier will revoke if msg.sender is not authorized.
+    modifier auth {
+        require(wards[msg.sender] == 1, "DssDirectDepositHub/not-authorized");
+        _;
+    }
+
+    /// @notice Mutex to prevent reentrancy on external functions
+    modifier lock {
+        require(locked == 0, "DssDirectDepositHub/system-locked");
+        locked = 1;
+        _;
+        locked = 0;
+    }
+
+    // --- Math ---
+    uint256 internal constant RAY = 10 ** 27;
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+    function _max(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x >= y ? x : y;
+    }
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x + (y - 1)) / y;
+    }
+
+    // --- Administration ---
+    /**
+        @notice Makes an address authorized to perform auth'ed functions.
+        @dev msg.sender must be authorized.
+        @param usr address to be authorized
+    */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+        @notice De-authorizes an address from performing auth'ed functions.
+        @dev msg.sender must be authorized.
+        @param usr address to be de-authorized
+    */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
```

Comment:
```suggestion
    function deny(address usr) external auth {
        require(vat.live() == 1, "DssDirectDepositHub/no-deny-during-shutdown");
        wards[usr] = 0;
        emit Deny(usr);
    }
```

Reply:
Same here, governance can't `deny` anything that will make this module to break.